stack/core/src/main/java/org/apache/usergrid/corepersistence/CpEntityManager.java

```diff
@@ -546,38 +550,32 @@ public void update( Entity entity ) throws Exception {
             handleWriteUniqueVerifyException( entity, wuve );
         }
 
-        if ( !skipIndexingForType( cpEntity.getId().getType() ) ) {
+        if (!skipIndexingForType) {
+            indexEntity(cpEntity, asyncIndex);
+            deIndexOldVersionsOfEntity(cpEntity);
+        }
+    }
 
-            // queue an event to update the new entity
-            indexService.queueEntityIndexUpdate( applicationScope, cpEntity, 0 );
+    private void indexEntity(org.apache.usergrid.persistence.model.entity.Entity cpEntity, Boolean async) {
+        // queue an event to update the new entity
+        indexService.queueEntityIndexUpdate( applicationScope, cpEntity, 0 , async);
+    }
 
-            // queue up an event to clean-up older versions than this one from the index
-            if (entityManagerFig.getDeindexOnUpdate()) {
-                indexService.queueDeIndexOldVersion( applicationScope, cpEntity.getId(), cpEntity.getVersion());
-            }
+    private void deIndexOldVersionsOfEntity(org.apache.usergrid.persistence.model.entity.Entity cpEntity) {
```

Comment:
It would otherwise be confused with org.apache.usergrid.persistence.Entity which is brought in by the "import org.apache.usergrid.persistence.*;" and used all over the file.

I

Reply:
ok, sounds good